### ISSUE TITLE
(Pro) No module named 'fbs_runtime.application_context' when running app that was released via buildvm/runvm

### ISSUE THREAD

**Issue by apastel** [mherrmann/fbs#294]
My application runs fine when using the `fbs freeze` target executable, but when running the app that was released using the Ubuntu VM, the application silently fails to start. When attempting to start it by running the executable in the terminal, the following error is shown

```
$ /opt/HourlyTracker/HourlyTracker 
Traceback (most recent call last):
  File "main.py", line 14, in <module>
ModuleNotFoundError: No module named 'fbs_runtime.application_context'
[62935] Failed to execute script main
```

As shown in the manual, my application context is instantiated in `main.py` as follows:
```
from fbs_runtime.application_context.PySide6 import ApplicationContext

if __name__ == "__main__":
    appctxt = ApplicationContext()
    window = MainWindow()
```

I'm not sure what it is about the Linux release process that is preventing the fbs_runtime module from being included, but I think it's worth it for me to share my `base.txt` which will highlight some quirks I found with the release process that could be clues for my problem.
```
appdirs
boto3
fbs
#./fbs_pro-1.1.3.tar.gz
PySide6
flake8
flake8-bugbear
flake8-implicit-str-concat
```
Most notable here is the commented-out fbs_pro tarball and the inclusion of the public `fbs` instead. Ideally, I would only be using the `fbs_pro` file instead of `fbs`. While the app has no problem with `fbs freeze` using the fbs_pro file, when I attempt to run `fbs buildvm ubuntu` I was getting `No such file or directory: '/root/HourlyTracker/fbs_pro-1.1.3.tar.gz'`. This is presumably because buildvm only copies the base.txt file into the VM, and not the actual fbs_pro file. So during the pip install inside the VM, it fails to find the fbs_pro file. My workaround is to just add `fbs` as a dependency when I run buildvm and comment out the pro file. This seems wrong to have to do so it's possible that is a source of the problem. I'm not sure how else to make it work though -- I've reviewed the Release instructions many times.

Less notable is the addition of `boto3` which I added because otherwise `fbs release` inside the VM was giving
```
File "/root/.pyenv/versions/3.6.12/lib/python3.6/site-packages/fbs/_aws.py", line 19, in upload_file
    import boto3
ModuleNotFoundError: No module named 'boto3'
```

I tried to see if `fbs freeze --debug` provided some clue but the only output shown is the "Done. You can now run 'target/appname/appname" message.

* fbs pro version: 1.1.3
* Host OS: Linux Mint (based on Ubuntu 20.04)
* PySide6

**Comment by mherrmann** [comment]
Yes, in that case you need to install Pro inside the Linux VM. So in your case, you need to somehow get fbs_pro.tar.gz into the Linux VM so it gets used by `base.txt`. You'd need to remove the `fbs` line there and only keep the `.tar.gz` line. fbs Pro should have a ready solution for this, at least in the manual. But I'm afraid I can't offer to add this for the Hobbyist license you purchased, which comes without support.

**Comment by apastel** [comment]
If this is the manual, there doesn't seem to be a documented solution for this: https://build-system.fman.io/manual/

Just wondering how other Pro users are doing this (without having to hack the Dockerfiles inside fbs_pro.tar.gz). How is it designed to work out of the box?

**Comment by meramsey** [comment]
put file in private url and then reference that in your requirements.txt

```
fbs @ https://somedomain/somesecretpath/fbs_pro-1.1.8.tar.gz
```

or you would need to volume mount the tarball inside and do a local install

**Comment by apastel** [comment]
Thanks @meramsey, both options work. I went with the latter by modifying the `_defaults/src/build/docker/ubuntu/Dockerfile` inside the fbs pro tarball to add this line:
```
COPY ./fbs_pro-1.1.3-apastel.tar.gz /root/${app_name}/
```
with that tarball residing locally in my requirements folder, and the requirements.txt file listing it as `./fbs_pro-1.1.3-apastel.tar.gz`.
I went with that approach because I ended up having to modify that Dockerfile already anyway to fix a different bug. Had to change
```
ARG python_version=3.6.12
```
to
```
ARG python_version=3.7.0
```
because Python 3.6 is not compatible with a line in `fbs/freeze/__init__.py:48` that uses the `capture_output=True` keyword argument in the line
```
cp = run(args, capture_output=True, text=True)
```

Now I'm able to build the VM, create the release, and install and run the app without any errors that are specific to fbs. (Just errors pertaining to Qt that could be separate.) It's still a little puzzling though that hacking open the tarball or hosting it on the internet somewhere is some undocumented required solution to get the Linux build working, especially considering this is not free software (did everyone have to do this same hack?). But I was able to successfully release a separate Windows application just fine at least, so it was still worth it for the Windows portion.

**Comment by mherrmann** [comment]
I fixed these issues now in releases [1.2.0](https://github.com/mherrmann/fbs/releases/tag/v1.2.0) and [1.2.1](https://github.com/mherrmann/fbs/releases/tag/v1.2.1). If your purchase was within the last 12 months and you let me know the email you purchased with, then I'll give you access to the new version.

**Comment by apastel** [comment]
Fantastic, much appreciated. I forwarded my purchase receipt from August '22 to your email address.
